javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/collector/ReferenceCollector.java

```diff
@@ -201,6 +203,73 @@ private void collectHelperClasses(
     return references;
   }
 
+  public void prune() {
```

Comment:
Can you add some tests for this method?

Reply:
I still need to do this 👍

Reply:
I added `prune` to all of the existing tests in `ReferenceCollectorTest`, which I think is a good way of showing that `prune` doesn't remove anything important, let me know if there is some other coverage you would like to see

Reply:
you (rightly) don't sound very convinced 😂, so I have opened #2962 and assigned it to myself 👍